src/Attestation/Attestation/Models/AttestationDataServiceClient.cs

```diff
@@ -163,10 +164,28 @@ private string GetDataPlaneUri(string name, string resourceGroupName, bool refre
 
         private void ThrowOn4xxErrors(AzureOperationResponse<object> result)
         {
-            if (result.Response.StatusCode == HttpStatusCode.BadRequest)
-                throw new ArgumentException($"Operation returns BadRequest");
-            if (result.Response.StatusCode == HttpStatusCode.Unauthorized)
-                throw new ArgumentException("Operation is unauthorized");
+            int statusCode = (int) result.Response.StatusCode;
+
+            if (statusCode >= 400 && statusCode <= 499)
+            {
+                var responseBody = result.Response.Content.ReadAsStringAsync().Result;
+                var errorDetails = $"Operation returned HTTP Status Code {statusCode}";
+
+                // Include response body as either parsed ServerError or string
+                if (!string.IsNullOrEmpty(responseBody))
+                {
+                    try
+                    {
+                        var error = JsonConvert.DeserializeObject<ServerError>(responseBody).Error;
+                        errorDetails += $"\n\rCode: {error.Code}\n\rMessage: {error.Message}\n\r";
+                    }
+                    catch (Exception)
+                    {
+                        errorDetails += $"\n\rResponse Body: {responseBody}\n\r";
+                    }
+                }
+                throw new RestException(errorDetails);
```

Comment:
I suppose test case need to be updated if there is one and is expecting the "ArgumentException"

Reply:
Good question.  There are no test cases specifically looking for the value "ArgumentException".  All tests are passing.